plugins/PlatformFSPlugin.ts

```diff
@@ -0,0 +1,233 @@
+import { parse as parseFile, join, basename, relative } from "path";
+import * as minimatch from "minimatch";
+
+export interface PlatformFSPluginOptions {
+    /**
+     * The target platform.
+     */
+    platform?: string;
+
+    /**
+     * A list of all platforms. By default it is `["ios", "android"]`.
+     */
+    platforms?: string[];
+
+    /**
+     * An array of minimatch expressions used to filter nodes from the file system.
+     */
+    ignore?: string[];
+}
+
+export class PlatformFSPlugin {
+    protected readonly platform: string;
+    protected readonly platforms: ReadonlyArray<string>;
+    protected readonly ignore: ReadonlyArray<string>;
+    protected context: string;
+
+    constructor({ platform, platforms, ignore }: PlatformFSPluginOptions) {
+        this.platform = platform || "";
+        this.platforms = platforms || ["ios", "android"];
+        this.ignore = ignore || [];
+    }
+
+    public apply(compiler) {
+        this.context = compiler.context;
+        compiler.inputFileSystem = mapFileSystem({
+            fs: compiler.inputFileSystem,
+            platform: this.platform,
+            platforms: this.platforms,
+            ignore: this.ignore,
+            context: this.context
+        });
+    }
+}
+
+export interface MapFileSystemArgs {
+    /**
+     * This is the underlying webpack compiler.inputFileSystem, its interface is similar to Node's fs.
+     */
+    readonly fs: any;
+    readonly context: string;
+    readonly platform: string;
+    readonly platforms: ReadonlyArray<string>;
+    readonly ignore: ReadonlyArray<string>;
+}
+
+export function mapFileSystem(args: MapFileSystemArgs): any {
+    let { fs, platform, platforms, ignore, context } = args;
+    ignore = args.ignore || [];
+
+    const minimatchFileFilters = ignore.map(pattern => {
+        const minimatchFilter = minimatch.filter(pattern);
+        return file => minimatchFilter(relative(context, file));
+    });
+
+    const isIgnored = file => minimatchFileFilters.some(filter => filter(file));
+
+    const alienPlatforms = platforms.filter(p => p !== platform);
+    const alienPlatformFilters = alienPlatforms.map(platform => ({
+            endsWithSuffix: `.${platform}`,
+            contains: `.${platform}.`
```

Comment:
This may fire false positives on:
component.android.ts // ok
component.android.myapp.ts // not-ok